frontend/src/containers/PostDetail.tsx

```diff
@@ -0,0 +1,213 @@
+import {State} from "reducers/index";
+import {fetchPostComments, fetchPostDetails, PostDetail} from "actions/posts";
+import {connect} from "react-redux";
+import React, {useEffect, useState} from "react";
+import {Icon, List} from "antd";
+import {ApiComment, CommentData, submitComment} from "actions/comments";
+import {withPolling} from "src/polling";
+import CreateCommentForm from "components/CommentForm";
+import {Web3Context} from "reducers/bc";
+import {initWeb3} from "actions/bc";
+import {submitCommentVote, submitPostVote} from "actions/votes";
+
+
+import 'src/styles/PostsDetail.scss';
+
+interface StateToProps {
+    post: PostDetail,
+    comments: ApiComment[] | undefined,
+    web3Context: Web3Context | null,
+    commentSubmit: {
+        loading: boolean,
+        error: Error | null,
+    },
+    voteSubmit: {
+        loading: boolean,
+        error: Error | null,
+    }
+}
+
+interface DispatchToProps {
+    fetchPostComments: typeof fetchPostComments,
+    fetchPostDetails: typeof fetchPostDetails,
+    submitComment: typeof submitComment,
+    initWeb3: typeof initWeb3,
+    submitPostVote: typeof submitPostVote,
+    submitCommentVote: typeof submitCommentVote
+}
+
+// TODO probably should use RouteComponentProps
```

Comment:
yup, take a look at `RouteComponentProps` interface, it accepts optional Params interface